chain/chain/src/chain.rs

```diff
@@ -382,12 +382,18 @@ impl Chain {
         self.doomslug_threshold_mode = DoomslugThresholdMode::NoApprovals
     }
 
-    pub fn compute_bp_hash_inner(bps: &Vec<ValidatorStake>) -> Result<CryptoHash, Error> {
-        let mut arr = vec![];
-        for bp in bps.iter() {
-            arr.append(&mut hash(bp.account_id.as_bytes()).into());
-            arr.append(&mut hash(bp.public_key.try_to_vec()?.as_ref()).into());
-            arr.append(&mut hash(bp.stake.try_to_vec()?.as_ref()).into());
+    pub fn compute_bp_hash_inner<'a, T>(bps: T) -> Result<CryptoHash, Error>
+    where
+        T: IntoIterator<Item = &'a ValidatorStake>,
+    {
+        let iter = bps.into_iter();
+        let (size, _) = iter.size_hint();
+        // each element contributes three 32-byte hashes
+        let mut arr = Vec::with_capacity(96 * size);
```

Comment:
Let’s use `std::mem::size_of::<CryptoHash>() * 3` instead of the hardcoded const

Reply:
💯 
https://github.com/nearprotocol/nearcore/pull/2634/commits/4ca95be9497fd71d243cf5addf98027c17959db8